### PR TITLE
NO-ISSUE: fix(assisted-installer-agent): Prepare replacing ci test images

### PR DESCRIPTION
**Pull request by pastequo**
Fixing prow build for 2.8 & 2.9 ocm releases

Prerequisite for 
* https://github.com/openshift/assisted-installer-agent/pull/742
* https://github.com/openshift/assisted-installer-agent/pull/743

/cc @danmanor

### CODE REVIEW

File: ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-release-ocm-2.8.yaml [hunk 1]
```diff
@@ -120,29 +90,37 @@ tests:
     - ref: assisted-baremetal-images-publish
 - as: lint
   commands: |
+    export GOCACHE=/tmp/gocache
+    export GOMODCACHE=/tmp/gomodcache
     export GOLANGCI_LINT_CACHE=/tmp/.cache
     export GOPROXY=https://proxy.golang.org
-    export "GOROOT=$(go env GOROOT)"
```
Comment (pastequo) [inline ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-release-ocm-2.8.yaml:125]: @danmanor I guess it's safe to remove ?

File: ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-release-ocm-2.8.yaml [hunk 2]
```diff
@@ -73,9 +71,6 @@ resources:
     requests:
       cpu: 100m
       memory: 200Mi
-test_binary_build_commands: |
```
Comment (pastequo) [inline ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-release-ocm-2.8.yaml:76]: @danmanor I removed those lines to align with other releases. Not sure if it's required

File: ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-release-ocm-2.9.yaml [hunk 3]
```diff
@@ -73,9 +71,6 @@ resources:
     requests:
       cpu: 100m
       memory: 200Mi
-test_binary_build_commands: |
-  make build-openshift-ci-test-bin
-  go mod vendor
```
Comment (pastequo) [inline ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-release-ocm-2.9.yaml:78]: @danmanor I removed those lines to align with other releases. Not sure if it's required

File: ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-release-ocm-2.8.yaml [hunk 4]
```diff
@@ -23,64 +23,35 @@ base_images:
...
     name: release
     namespace: openshift
     tag: golang-1.19
 build_root:
-  project_image:
-    dockerfile_path: Dockerfile.assisted_installer_agent-build
+  from_repository: true
 images:
 - dockerfile_path: Dockerfile.assisted_installer_agent
-  from: coreos-centos-stream8
```
Comment (danmanor) [inline ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-release-ocm-2.8.yaml:47]: Lets leave Dockerfiles that don't end with "-build" for now

File: ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-release-ocm-2.8.yaml [hunk 5]
```diff
@@ -23,64 +23,35 @@ base_images:
     name: ocm-2.8
     namespace: edge-infrastructure
     tag: assisted-service-scripts
-  coreos-centos-stream8:
```
Comment (danmanor) [inline ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-release-ocm-2.8.yaml:26]: Lets leave Dockerfiles that don't end with "-build" for now

File: ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-release-ocm-2.9.yaml [hunk 6]
```diff
@@ -23,10 +23,10 @@ base_images:
     name: ocm-2.9
     namespace: edge-infrastructure
     tag: assisted-service-scripts
-  coreos-centos-stream8:
```
Comment (danmanor) [inline ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-release-ocm-2.9.yaml:26]: Same comment

File: ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-release-ocm-2.9.yaml [hunk 7]
```diff
@@ -36,16 +36,17 @@ base_images:
     namespace: openshift
     tag: golang-1.19
 build_root:
-  project_image:
-    dockerfile_path: Dockerfile.assisted_installer_agent-build
+  from_repository: true
 images:
 - dockerfile_path: Dockerfile.assisted_installer_agent
-  from: coreos-centos-stream8
```
Comment (danmanor) [inline ci-operator/config/openshift/assisted-installer-agent/openshift-assisted-installer-agent-release-ocm-2.9.yaml:43]: Same comment

**Review by danmanor — CHANGES_REQUESTED** [review]
we also need to change `assisted-installer-agent-build` in ocm-2.9, 2.10, master__edge

**Comment by pastequo** [comment]
> we also need to change assisted-installer-agent-build in ocm-2.9, 2.10, master__edge

I guess you meant ocm-2.10 2.11 and master__edge ?

**Comment by danmanor** [comment]
LGTM

**Pull request merged**